### ISSUE TITLE
Support: Dynamically adding fields to existing form

### ISSUE THREAD

**Issue by davidgustys** [foxhound87/mobx-ajv-form#369]
I have an issue or most likely I don't know how to accomplish this, i have read the documentation and also looked into the source code, but couldn't find a nice straight way to do it.

My problem is that i have a Form which field list might/is growing based on user input. At first a tried using `Form.init` to basically re-initialise the form with new fields, but as one would expect, the existing Form got *reset* and thus state changed causing unwanted UI re-render. 

Then I tried `Form.add('', {key})` combined with `For.set` - that does seem to work and meet my needs, but logic is not elegant since im passing the same object that would passed to Form.init, thus i need to use loops to pass data into those methods. 

So basically I would to know if there is better way of doing this, maybe im approaching this situation from the wrong direction? 

For me it seems if I had something like `Form.init(fields, {reset: false})` would be the simplest approach, but then again maybe I missed something.

All in all the library seems quite nice, thank you for sharing and the support !

**Comment by ekozhura** [comment]
Please, take a look at this ticket: https://github.com/foxhound87/mobx-react-form/issues/360
For me, `add` method didn't even work properly, so i switched to `initField`. `Form.init` uses `initFields` inside, which iterates over ypur fields object and calls `initField` on each field object.

**Comment by davidgustys** [comment]
@ekozhura yeah i've seen that method, but I guess it was only intended to be used as a private function rather than public. Anyways i've changed the flow of my application form creation and now i'm able to use this library. 

Speaking more generally thought it would be nice to have more utility methods. But looking at the activity from the creator/maintainer its not clear if we will get them anytime soon.

**Comment by ekozhura** [comment]
@gdeividas it's really hard to tell which methods are considered to be a part of public API and which are not.

**Comment by foxhound87** [comment]
I've reimplemented the `add()` method in the version `1.34`. Try it.
